libraries/AP_RangeFinder/AP_RangeFinder_Benewake.cpp

```diff
@@ -107,25 +108,31 @@ bool AP_RangeFinder_Benewake::get_reading(uint16_t &reading_cm, bool &signal_ok)
                 }
                 // if checksum matches extract contents
                 if (checksum == linebuf[BENEWAKE_FRAME_LENGTH-1]) {
+                    // tell caller we are receiving packets
+                    signal_ok = true;
                     // calculate distance and add to sum
-                    uint16_t dist = be16toh(((uint16_t)linebuf[2] << 8) | linebuf[3]);
-                    // TFmini has short distance mode (mm)
-                    if ((model_type == BENEWAKE_TFmini)) {
-                        if (linebuf[6] == 0x02) {
-                            dist *= 10;
+                    uint16_t dist = le16toh(((uint16_t)linebuf[3] << 8) | linebuf[2]);
```

Comment:
I did a terrible job explaining here, sorry. We have two options:

- we either do what's between parenthesis: `((uint16_t)linebuf[3] << 8) | linebuf[2]);`, OR
- we call le16toh with a le16_t value: `le16toh(*((le16_t*)&linebuf[2]))`

Doing both operations isn't needed. The second option is more useful where packets are defined by a packed structure and a buffer is just copied into it, here the first option seems the better.

Reply:
Ok, I don't know what to change.  It works at the moment but if I flip around [3] and [2] it doesn't work...

Reply:
It should be `uint16_t dist = ((uint16_t)linebuf[3] << 8) | linebuf[2]; `. There is no point in calling the method because we are already doing the operation to make it correct.

Reply:
@OXINARF, cool thanks.  I've pushed fixes now.  We should squash all these together before merging probably.  If you give the LGTM I can do that..